blocks/hooks/generated-class-name.js

```diff
@@ -25,11 +25,24 @@ import { hasBlockSupport, getBlockDefaultClassname } from '../api';
 export function addGeneratedClassName( extraProps, blockType ) {
 	// Adding the generated className
 	if ( hasBlockSupport( blockType, 'className', true ) ) {
-		const updatedClassName = classnames(
-			getBlockDefaultClassname( blockType.name ),
-			extraProps.className,
-		);
-		extraProps.className = updatedClassName;
+		const blockDefaultClassname = getBlockDefaultClassname( blockType.name );
+
+		const blockDefaultClassnameIsDupe = ( typeof( extraProps.className ) === 'string' &&
```

Comment:
`typeof` is a keyword, not a function, and we should probably omit the parentheses to prevent confusion to this point:

```js
typeof extraProps.className
```

Reply:
Minor: On variable naming, I tend to encourage prefixing the `is` to indicate a boolean value at a glance:

```js
const isBockDefaultClassnameDupe = /* ... */;
```